### PR TITLE
Disallow duplicate dimension sets

### PR DESCRIPTION
**Pull request by Stephen-Bao**
*Issue #, if available:*
#83 

*Description of changes:*
This is the fix for preventing duplicate dimension set in python library associated with related issues in other libraries described in #83.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws_embedded_metrics/logger/metrics_context.py [hunk 1]
```diff
@@ -79,6 +79,12 @@ def put_dimensions(self, dimensions: Dict[str, str]) -> None:
 
         self.validate_dimension_set(dimensions)
 
+        # Duplicate dimension sets are removed before being added to the end of the collection.
+        # This ensures only latest dimension value is used as a target member on the root EMF node.
+        # This operation is O(n^2), but acceptable given sets are capped at 30 dimensions
+        incoming_keys: Set = set(dimensions.keys())
+        self.dimensions = list(filter(lambda dim: (set(dim.keys()) != incoming_keys), self.dimensions))
```
Comment (markkuhn) [inline aws_embedded_metrics/logger/metrics_context.py:86]: The following test doesn't pass:

```python
    context = MetricsContext()
    dimension1 = {'key1': 'val1'}
    dimension2 = {'key2': 'val2'}

    context.set_dimensions(dimension1)
    context.put_dimensions(dimension2)

    assert len(context.get_dimensions()) == 2
```

**Review by markkuhn — CHANGES_REQUESTED** [review]

**Review by gordonpn — CHANGES_REQUESTED** [review]
Can we have another test that calls `set_dimensions()` in between two `put_dimensions()` calls and verifying the length/values?

```
put_dimensions()
set_dimensions()
put_dimensions()
```

File: aws_embedded_metrics/logger/metrics_context.py (hunk 1 above)
Comment (Stephen-Bao) [reply aws_embedded_metrics/logger/metrics_context.py]: Solved. Should be `context.set_dimensions([dimension1])`.

**Review by gordonpn — APPROVED** [review]

**Review by markkuhn — APPROVED** [review]

**Comment by markkuhn** [comment]
🚀

**Review by markkuhn — APPROVED** [review]

**Pull request merged**